可否支持IronPython库

IronPython库作用是可.Net和Python之间的互相调用,
而IronPython库感觉像是独立的Python环境, 但IronPython与Python2.7却有一些差异
主要原因是IronPython中处理字符串均是unicode, 看以下代码:
```
# IronPython 2.7.7 (2.7.7.0) on .NET 4.0.30319.42000 (32-bit)
# Type "help", "copyright", "credits" or "license" for more information.
import pypinyin
from pypinyin import pinyin
pinyin('成都')
#报出以下错误
#Traceback (most recent call last):
#  File "<stdin>", line 1, in <module>
#  File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\pypinyin\core.py", line 250, in pinyin
#  File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\pypinyin\core.py", line 32, in seg
#  File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\pypinyin\core.py", line 34, in seg
#  File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\pypinyin\utils.py", line 48, in simple_seg
#AssertionError: must be unicode string or [unicode, ...] list

########################################
print unicode   #output <type 'str'>
print str       #output <type 'str'>
# 从上面两行代码可以看出, 在IronPython环境中, unicode与str完全是相同的
print sys.version_info
#output sys.version_info(major=2, minor=7, micro=7, releaselevel='final', serial=0)
print sys.subversion
#output ('IronPython', '', '')
```
从上面可以看出 在IronPython输出的版本是2.7.7.0, 在你compat.py中判定`PY2`变量为`True`, 所以`bytes_type`的结果为`<type str>`
由于在IronPython中输出`unicode`与`str`是都是`str`, 所以在`utils.py`对`hans`判断为`bytes_type`时便会报`must be unicode string or [unicode, ...] list`
希望可以在`compat.py`文件对`PY2`变量加以下判断
```
PY2 = sys.version_info < (3, 0) and sys.subversion[0] != 'IronPython'
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,6 +1,12 @@
 Changelog
 ---------
 
+0.22.0 (2017-mm-dd)
++++++++++++++++++++++
+
+* **[New]** 支持 IronPython (via `#xx`_). Thanks `@LevyLession`_
+
+
 0.21.1 (2017-05-29)
 +++++++++++++++++++++
 
@@ -475,3 +481,5 @@ __ https://github.com/mozillazg/python-pinyin/issues/8
 .. _#59: https://github.com/mozillazg/python-pinyin/issues/59
 .. _#72: https://github.com/mozillazg/python-pinyin/issues/72
 .. _phrase-pinyin-data: https://github.com/mozillazg/phrase-pinyin-data
+.. _@LevyLession: https://github.com/LevyLession
+.. _#xx: https://github.com/mozillazg/python-pinyin/issues/xx

--- a/pypinyin/compat.py
+++ b/pypinyin/compat.py
@@ -8,6 +8,14 @@ import sys
 SUPPORT_UCS4 = len('\U00020000') == 1
 
 PY2 = sys.version_info < (3, 0)
+
+subversion = getattr(sys, 'subversion', [''])
+# 这些 Python 实现虽然是 Python 2 但字符串的行为跟 Python 3 是一样的
+if subversion[0] in (
+    'IronPython',
+):
+    PY2 = False
+
 if not PY2:
     text_type = str
     bytes_type = bytes

